### ISSUE TITLE
Navigating through Waypoints stops at first coordinate

### ISSUE THREAD

**Issue by LukasStancikas** [mapbox/mapbox-navigation-android#27]
Hi, i have implemented a simple navigation UI, from the information that "RouteProgress" object gives. this is how i create "MapboxDirection" object:
  ```
   return new MapboxDirections.Builder()
                .setOverview(DirectionsCriteria.OVERVIEW_FULL)
                .setSteps(true)
                .setAccessToken(getString(R.string.mapbox_access_token))
                .setCoordinates(coords)
                .setProfile(DirectionsCriteria.PROFILE_DRIVING)
                .build();
```

coords are a list of coordinates, where the first element in the list is my current location, the last element is the destination, and <21 points in between are waypoints (not stopovers). But the RouteProgress + onAlertLevelChange seems to think that when i come to my second coordinate (from my current location to first waypoint), that the trip is over and i get "You have arrived at your destination" in ```routeProgress.getCurrentLegProgress().getUpComingStep().getManeuver().getInstruction()```
and the ```routeProgress.getCurrentLegProgress().getDistanceRemaining()``` is 0. But ```routeProgress.getDistanceRemaining()``` and the duration of the whole routeProgress is still there. I also get ARRIVE_ALERT_LEVEL in onAlertLevelChange, at the second coordinate.So how do i make the navigation advance the route leg, without actually stopping navigatint after the fist waypoint.

**Comment by 1ec5** [comment]
Also tracking on iOS in mapbox/mapbox-navigation-ios#204.

**Comment by mecoFarid** [comment]
@LukasStancikas could you post code snippet (non-confidential parts), so I could implement it in my App. I'm unable to get it work as of problems listed https://github.com/mapbox/mapbox-navigation-android/issues/124 and https://github.com/mapbox/mapbox-navigation-android/issues/123. Thanks a lot.

**Comment by 1ec5** [comment]
Fixed in #111.
